pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -256,10 +253,16 @@ func (r *Reconciler) exists() (bool, error) {
 	if err := r.validateZone(); err != nil {
 		return false, fmt.Errorf("unable to verify project/zone exists: %v/%v; err: %v", r.projectID, zone, err)
 	}
-	_, err := r.computeService.InstancesGet(r.projectID, zone, r.machine.Name)
+	instance, err := r.computeService.InstancesGet(r.projectID, zone, r.machine.Name)
 	if err == nil {
-		klog.Infof("Machine %q already exists", r.machine.Name)
-		return true, nil
+		switch instance.Status {
+		case "PROVISIONING", "REPAIRING", "RUNNING", "STAGING":
```

Comment:
I wonder whether this should be inverted: 

```go
		switch instance.Status {
		case "TERMINATED":
			klog.Infof("Machine %q is considered as non existent as its status is %q", instance.Status)
			return false, nil
		default:
			klog.Infof("Machine %q already exists", r.machine.Name)
			return true, nil
		}

```

Reply:
we should make all explicit and error otherwise

Reply:
> we should make all explicit and error otherwise

This becomes brittle. We only really care for the `TERMINATED`.